File: SPRING/Hospital_Management_system_angular/src/app/services/billing/payment.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../../environments/environment';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class PaymentService {
  private api = environment.apiUrl + 'payments';
  constructor(private http: HttpClient) {}
  getAll(): Observable<any[]> { return this.http.get<any[]>(this.api); }
  getById(id: number): Observable<any> { return this.http.get<any>(`${this.api}/${id}`); }
  getByInvoice(invoiceNumber: string): Observable<any[]> { return this.http.get<any[]>(`${this.api}/invoice/${invoiceNumber}`); }
  getByPatient(patientId: number): Observable<any[]> { return this.http.get<any[]>(`${this.api}/patient/${patientId}`); }
  processPayment(data: any): Observable<any> { return this.http.post<any>(this.api, data); }
  updatePayment(id: number, data: any): Observable<any> { return this.http.put<any>(`${this.api}/${id}`, data); }
  deletePayment(id: number): Observable<any> { return this.http.delete<any>(`${this.api}/${id}`); }
  getDashboardStats(): Observable<any> { return this.http.get<any>(`${this.api}/dashboard-stats`); }
  getDailyRevenue(date: string): Observable<any> { return this.http.get<any>(`${this.api}/daily-revenue`, { params: { date } }); }
  getMonthlyRevenue(year: number, month: number): Observable<any> { return this.http.get<any>(`${this.api}/monthly-revenue`, { params: { year, month } }); }
  getMethodBreakdown(): Observable<any[]> { return this.http.get<any[]>(`${this.api}/method-breakdown`); }
}

File: Android_Hospital_Management/Modern_Hospital_Management_System-Backend/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/PaymentService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.entity.Payment;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;

public interface PaymentService {

    List<Payment> getAll();

    Payment getById(Long id);

    List<Payment> getByInvoiceNumber(String invoiceNumber);

    List<Payment> getByPatientId(Long patientId);

    Payment processPayment(Payment payment);

    Payment updatePayment(Payment payment);

    void deletePayment(Long id);

    Map<String, Object> getDashboardStats();

    Map<String, Object> getDailyRevenue(LocalDate date);

    Map<String, Object> getMonthlyRevenue(int year, int month);

    List<Map<String, Object>> getPaymentMethodBreakdown();

    List<Payment> getUnpaidByPatientId(Long patientId);

    List<Payment> getUnpaidByPatientName(String patientName);

    List<Payment> searchUnpaid(String search);
}
